src/StakingNode.sol

```diff
@@ -246,15 +330,135 @@ contract StakingNode is IStakingNode, StakingNodeEvents, ReentrancyGuardUpgradea
+    /// @param middlewareTimesIndexes The middlewareTimesIndex parameter has to do
+    ///       with the Slasher, which currently does nothing. As of M2, this parameter
+    ///       has no bearing on anything and can be ignored
+    function completeQueuedWithdrawals(
+        IDelegationManager.Withdrawal[] memory withdrawals,
+        uint256[] memory middlewareTimesIndexes
+        ) external onlyOperator {
+
+        uint256 totalWithdrawalAmount = 0;
+
+        bool[] memory receiveAsTokens = new bool[](withdrawals.length);
+        IERC20[][] memory tokens = new IERC20[][](withdrawals.length);
+        for (uint256 i = 0; i < withdrawals.length; i++) {
+
+            // Set receiveAsTokens to true to receive ETH when completeQueuedWithdrawals runs.
+            ///IMPORTANT: beaconChainETHStrategy shares are non-transferrable, so if `receiveAsTokens = false`
+            // and `withdrawal.withdrawer != withdrawal.staker`, any beaconChainETHStrategy shares
+            // in the `withdrawal` will be _returned to the staker_, rather than transferred to the withdrawer,
+            // unlike shares in any other strategies, which will be transferred to the withdrawer.
+            receiveAsTokens[i] = true;
+
+            // tokens array must match length of the withdrawals[i].strategies
+            // but does not need actual values in the case of the beaconChainETHStrategy
+            tokens[i] = new IERC20[](withdrawals[i].strategies.length);
+
+            for (uint256 j = 0; j < withdrawals[i].shares.length; j++) {
+                totalWithdrawalAmount += withdrawals[i].shares[j];
+            }
         }
+
+        IDelegationManager delegationManager = IDelegationManager(address(stakingNodesManager.delegationManager()));
+
+        uint256 initialETHBalance = address(this).balance;
+
+        // NOTE:  completeQueuedWithdrawals can only be called by withdrawal.withdrawer for each withdrawal
+        // The Eigenlayer beaconChainETHStrategy  queued withdrawal completion flow follows the following steps:
+        // 1. The flow starts in the DelegationManager where queued withdrawals are managed.
+        // 2. For beaconChainETHStrategy, the DelegationManager calls _withdrawSharesAsTokens interacts with the EigenPodManager.withdrawSharesAsTokens
+        // 3. Finally, the EigenPodManager calls withdrawRestakedBeaconChainETH on the EigenPod of this StakingNode to finalize the withdrawal.
+        // 4. the EigenPod decrements withdrawableRestakedExecutionLayerGwei and send the ETH to address(this)
+        delegationManager.completeQueuedWithdrawals(withdrawals, tokens, middlewareTimesIndexes, receiveAsTokens);
+
+        uint256 finalETHBalance = address(this).balance;
+        uint256 actualWithdrawalAmount = finalETHBalance - initialETHBalance;
+        if (actualWithdrawalAmount != totalWithdrawalAmount) {
+            revert MismatchInExpectedETHBalanceAfterWithdrawals();
```

Comment:
seems like something which might happen due to how complex eigenlayer is and such. Therefore lets include the actual and expected ETH amounts (so 2 event args) here to make it easier to debug in case this happens!